app/controllers/web_rake/tasks_controller.rb

```diff
@@ -80,81 +79,53 @@ def ensure_environment_loaded
-      Dir.glob(Rails.root.join('lib/tasks/**/*.rake')).each do |file|
-        content = File.read(file)
-
-        # Match various task definition patterns
-
-        # Pattern for: task({ :sample_data => :environment })
-        content.scan(/task\s*\(\s*\{\s*:([a-zA-Z_][a-zA-Z0-9_]*)\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task({ sample_data: :environment }) - Ruby 1.9+ syntax with parens and braces
-        content.scan(/task\s*\(\s*\{\s*([a-zA-Z_][a-zA-Z0-9_]*)\s*:/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task :sample_data => :environment
-        content.scan(/task\s+:([a-zA-Z_][a-zA-Z0-9_]*)\s*=>/).each do |match|
-          task_names << match[0]
-        end
+      tasks = []
 
-        # Pattern for: task "sample_data" => :environment
-        content.scan(/task\s+["']([a-zA-Z_][a-zA-Z0-9_]*)["']\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task sample_data: :environment (Ruby 1.9+ syntax)
-        content.scan(/task\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*:/).each do |match|
-          task_names << match[0]
-        end
+      # Always include db:seed
+      if (seed_task = Rake::Task['db:seed'] rescue nil)
+        tasks << seed_task
+      end
 
-        # Pattern for: task(:sample_data)
-        content.scan(/task\s*\(\s*:([a-zA-Z_][a-zA-Z0-9_]*)\s*\)/).each do |match|
-          task_names << match[0]
-        end
+      # Find tasks defined in lib/tasks by checking their source location
+      Rake.application.tasks.each do |task|
+        next if task.actions.blank?
+        next if task.name == 'db:seed' # Already added above
 
-        # Pattern for: desc "..." \n task :name
-        content.scan(/desc\s+.*?\n\s*task\s+:([a-zA-Z_][a-zA-Z0-9_]*)/).each do |match|
-          task_names << match[0]
+        if task_from_lib_tasks?(task)
+          tasks << task
         end
       end
 
-      task_names.uniq
+      tasks
     end
 
     def find_task
       task_name = params[:id].gsub('__', ':')
 
-      # Check if this is one of our allowed tasks (custom tasks + db:seed)
-      allowed_tasks = extract_task_names_from_files + ['db:seed']
-      return nil unless allowed_tasks.include?(task_name)
-
       # Ensure tasks are loaded
       Rails.application.load_tasks
 
       # Find the task
-      Rake::Task[task_name] rescue nil
+      task = Rake::Task[task_name] rescue nil
+      return nil unless task
+
+      # Check if it's db:seed or defined in lib/tasks
+      return task if task_name == 'db:seed'
+      return task if task_from_lib_tasks?(task)
+
+      nil
+    end
+
+    def task_from_lib_tasks?(task)
+      return false if task.actions.blank?
+
+      source_path = task.actions.first.source_location&.first
+      source_path.present? &&
+        source_path.include?(Rails.root.to_s) &&
+        source_path.include?('/lib/tasks/')
```

Comment:
You can combine the path checks

```suggestion
        Rails.root.join("lib/tasks").to_s
```